File: Semester 4/Programming and Projection Enironments/mpp-proiect-java/Proiect_MPP/AirportPersistence/src/main/java/root/persistence/flights/IFlightRepository.java
package root.persistence.flights;

import root.model.Flight;
import root.persistence.Repository;

import java.time.LocalDateTime;
import java.util.Collection;

public interface IFlightRepository extends Repository<Flight, Integer> {
    Collection<Flight> getByDepartureAirport(int departureAirport);
    Collection<Flight> getByDestinationAirport(int destinationAirport);
    Collection<Flight> getAfterDepartureDateTime(LocalDateTime departureDateTime);
    Collection<Flight> getAvailable();
}

File: Semester 4/Programming and Projection Enironments/mpp-proiect-csharp/Proiect_MPP/persistence/flights/IFlightRepository.cs
using model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace persistence.flights
{
    public interface IFlightRepository : Repository<Flight, int>
    {
        List<Flight> getByDepartureAirport(int departureAirport);
        List<Flight> getByDestinationAirport(int destinationAirport);
        List<Flight> getAfterDepartureDateTime(DateTime departureDateTime);
        List<Flight> getAvailable();
    }
}
